core-bundle/src/Messenger/AutoFallbackWorker.php

```diff
@@ -0,0 +1,133 @@
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Messenger;
+
+use Psr\Cache\CacheItemInterface;
+use Psr\Cache\CacheItemPoolInterface;
+use Symfony\Component\Console\Input\ArrayInput;
+use Symfony\Component\Console\Output\NullOutput;
+use Symfony\Component\EventDispatcher\Attribute\AsEventListener;
+use Symfony\Component\HttpKernel\Event\TerminateEvent;
+use Symfony\Component\Messenger\Command\ConsumeMessagesCommand;
+use Symfony\Component\Messenger\Event\WorkerRunningEvent;
+use Symfony\Component\Messenger\Event\WorkerStartedEvent;
+use Symfony\Component\Messenger\Event\WorkerStoppedEvent;
+
+/**
+ * This service accepts an array of Symfony Messenger transports and automatically
+ * detects which of those have real workers running. In case they do not, it falls
+ * back to a kernel.terminate worker to ensure, messages are always processed, no
+ * matter whether a real worker is running or not.
+ *
+ * Detecting works by using the WorkerStartedEvent and WorkerRunningEvent which
+ * sets/updates a cache items and allowing a grace period.
+ *
+ * In any case, this provides advantages as it allows us to always use a queue and
+ * postpone processes to at least kernel.terminate in case there are no real
+ * workers available!
+ */
+class AutoFallbackWorker
+{
+    private const GRACE_PERIOD = 'PT60S'; // If no real worker "pings" us within this grace period, we run our fallback worker
+
+    private bool $fallbackWorkerRunning = false;
+
+    public function __construct(
+        private readonly CacheItemPoolInterface $cache,
+        private readonly ConsumeMessagesCommand $consumeMessagesCommand,
+        private readonly array $transports,
+    ) {
+    }
+
+    #[AsEventListener(priority: -512)]
+    public function onKernelTerminate(TerminateEvent $event): void
+    {
+        foreach ($this->transports as $transportName) {
+            $this->processTransport($transportName);
+        }
+    }
+
+    #[AsEventListener]
+    public function onWorkerStarted(WorkerStartedEvent $event): void
+    {
+        foreach ($event->getWorker()->getMetadata()->getTransportNames() as $transportName) {
+            $this->ping($transportName);
+        }
+    }
+
+    #[AsEventListener]
+    public function onWorkerRunning(WorkerRunningEvent $event): void
+    {
+        // In case of our fallback worker running, we stop it immediately if it is idle
+        // in order to free the web process.
+        if ($this->fallbackWorkerRunning && $event->isWorkerIdle()) {
+            $event->getWorker()->stop();
+
+            return;
+        }
+
+        foreach ($event->getWorker()->getMetadata()->getTransportNames() as $transportName) {
+            $this->ping($transportName);
+        }
+    }
+
+    #[AsEventListener]
+    public function onWorkerStopped(WorkerStoppedEvent $event): void
+    {
+        $this->fallbackWorkerRunning = false;
+    }
```

Comment:
Do we actually need this? We already set it to `false` on line 131 after `messenger:consume` is finished.
```suggestion
```

Reply:
Maybe we can remove it. It was more of like a `reset()` method to ensure it's always `false` after things ran.